### PR TITLE
fix(docs): batch markdown get operation for redis

### PR DESCRIPTION
**Pull request by dsinghvi**
## Short description of the changes made
Batches `mGet` at 100 mdx pages at a time. 

## What was the motivation & context behind this PR?
We don't want to exceed the redis quota. 

## How has this PR been tested?
Will be tested via preview/staging

### CODE REVIEW

**Review by abvthecity — APPROVED** [review]

File: packages/ui/fern-docs-mdx/src/sanitize/sanitize-mdx-expression.ts [hunk 1]
```diff
@@ -2,60 +2,81 @@ import { VFileMessage } from "vfile-message";
...
+    let loops = 0;
 
     // these are errors encountered but sanitized
-    const errors: VFileMessage[] = [];
+    const errors: ErrorContext[] = [];
 
     // eslint-disable-next-line no-constant-condition
     while (true) {
-        if (loops >= 100) {
+        if (loops++ >= 100) {
```
Comment (abvthecity) [inline packages/ui/fern-docs-mdx/src/sanitize/sanitize-mdx-expression.ts:37]: this is where we forgot to increment the for loop 🤦

File: packages/ui/fern-docs-mdx/src/sanitize/sanitize-mdx-expression.ts [hunk 2]
```diff
@@ -2,60 +2,78 @@ import { VFileMessage } from "vfile-message";
...
-
     // these are errors encountered but sanitized
-    const errors: VFileMessage[] = [];
+    const errors: ErrorContext[] = [];
 
-    // eslint-disable-next-line no-constant-condition
-    while (true) {
-        if (loops >= 100) {
+    let loops = 0;
+    while (loops++ < 100) {
```
Comment (abvthecity) [inline packages/ui/fern-docs-mdx/src/sanitize/sanitize-mdx-expression.ts:34]: this is where i forgot to close the for loop

File: packages/ui/fern-docs-mdx/src/sanitize/sanitize-mdx-expression.ts [hunk 3]
```diff
@@ -99,7 +127,9 @@ function handleUnexpectedEOF(content: string, _e: VFileMessage): [string, boolea
     let i = 0;
     while (i < content.length) {
         const char = content[i++];
-        if (char === "{" || char === "<") {
+        if (char === "\\") {
+            continue;
```
Comment (abvthecity) [inline packages/ui/fern-docs-mdx/src/sanitize/sanitize-mdx-expression.ts:131]: this was where it was bugging out

File: packages/ui/fern-docs-mdx/src/sanitize/sanitize-mdx-expression.ts (hunk 2 above)
Comment (dsinghvi) [reply packages/ui/fern-docs-mdx/src/sanitize/sanitize-mdx-expression.ts:34]: cool

**Pull request merged**